homeassistant/components/schlage/config_flow.py

```diff
@@ -41,13 +45,37 @@ async def async_step_user(
                 LOGGER.exception("Unknown error")
                 errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(user_id)
-                return self.async_create_entry(title=username, data=user_input)
+                if not self.reauth_entry:
```

Comment:
I think it would look cleaner to have these additions for reauth in `async_step_reauth_confirm`.
Not sure it gives any value with mixing it with the user step.